src/Microsoft.Health.Fhir.Core/Features/TaskManagement/TaskHosting.cs

```diff
@@ -0,0 +1,216 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Exceptions;
+
+namespace Microsoft.Health.Fhir.Core.Features.TaskManagement
+{
+    public class TaskHosting
+    {
+        private object _syncRoot = new object();
+
+        private ITaskConsumer _consumer;
+        private ITaskFactory _taskFactory;
+        private ILogger<TaskHosting> _logger;
+        private Dictionary<string, ITask> _activeTaskRecordsForKeepAlive = new Dictionary<string, ITask>();
```

Comment:
It might be possible to use a ConcurrentDictionary here to avoid the need of a lock object.

Reply:
Looks like it would not have conflict on same key write, updated to ConsurrentDictionary.